src/api/api.py

```diff
@@ -16,3 +20,186 @@ def test():
+    params = get_params_2()
+
+    r = requests.get(f"https://api.twitter.com/2/tweets?{tweet_ids}", headers=auth_header, params=params)
+    
+    return r.json()
+
+
+
+# Atualiza o banco de dados do twitter, colocando o username em deputados que possuem uma conta verificada
+@api.route('/update_twitter_accounts')
+def update_twitter_accounts():
+    for item in Deputy.objects:
+
+        #verificar se o edputado já tem seu twitter vinculado, se tiver, só passa pro próximo
+        if item.twitter_username or item.twitter_id:
+            continue
+
+        #criar as chaves para pesquisa
+        deputy_name = item.name
+        deputy_name = unidecode(deputy_name).replace(".", "").replace("'","").replace("-","").replace(" ", "")[0:13]
+
+        deputy_full_name = item.full_name
+        deputy_full_name = unidecode(deputy_full_name).replace(".", "").replace("'","").replace("-","").replace(" ", "")[0:13]
+
+        usernames_to_found = deputy_name + "," + deputy_full_name
+        r = requests.get(f"https://api.twitter.com/2/users/by?usernames={usernames_to_found}&user.fields=verified,url", headers=auth_header)
+
+        #caso a resposta seja nula, ignorar esse deputado
+        if not r:
+            print("Essa request não tem uma response") 
+            continue 
+
+        deputy_twitter_json = r.json()
+
+        # Criar uma String para conseguir verificar se ele possue a chave data
+        json_in_string = str(deputy_twitter_json)
+        
+        # if deputy_twitter_json.optString("data"):
+        if "data" in json_in_string:
+            #encontramos um deputado com conta
+
+            for all_usermaes_fouded in deputy_twitter_json["data"]:
+                if all_usermaes_fouded["verified"]:
+                    #conta verificada, adicionar no item
+                    item.twitter_username = all_usermaes_fouded["username"]
+                    item.twitter_id = all_usermaes_fouded["id"]
+                    
+                    if len(all_usermaes_fouded["url"]) > 5:
+                        item.website = all_usermaes_fouded["url"]
+                    
+                    item.save()
+                    break
+    
+    return "Done. Use url api/get_all_deputies for get all the deputies in db"
+
+
+# Rota que popula a classe Tweet com até 10 tweets mais recentes por usuario indicado pelo seu id
+@api.route('/update_tweets')
+def update_tweets():
+
+    params = get_params_2()
+
+    for deputy in Deputy.objects:
+        
+        if deputy.twitter_id:
+            r = requests.get(f"https://api.twitter.com/2/users/{deputy.twitter_id}/tweets", headers=auth_header, params=params)
+
+            if not r:
+                continue
+            
+            last_10_tweets_json = r.json()['data']
+
+            for tweet_json in last_10_tweets_json:
+                need_create = True
+                
+                for tweet_item in Tweet.objects:
```

Comment:
@fepas, Será que n tem um jeito mais eficiente de verificar isso? da uma olhada qnd for refatorar

Reply:
Isso é pro fepas, pode ignorar cristian.